src/ledger/LedgerManagerImpl.cpp

```diff
@@ -293,104 +294,117 @@ LedgerManagerImpl::loadLastKnownLedger(function<void()> handler)
-            missing.insert(missing.end(), pubmissing.begin(), pubmissing.end());
-            if (!missing.empty())
+            // Only restart merges in full startup mode. Many modes in core
+            // (standalone offline commands, in-memory setup) do not need to
+            // spin up expensive merge processes.
+            auto assumeStateWork =
+                mApp.getWorkScheduler().executeWork<AssumeStateWork>(
+                    has, latestLedgerHeader->ledgerVersion, restoreBucketlist);
+            if (assumeStateWork->getState() == BasicWork::State::WORK_SUCCESS)
             {
-                CLOG_ERROR(
-                    Ledger, "{} buckets are missing from bucket directory '{}'",
-                    missing.size(), mApp.getBucketManager().getBucketDir());
-                throw std::runtime_error("Bucket directory is corrupt");
+                CLOG_INFO(Ledger, "Assumed bucket-state for LCL: {}",
+                          ledgerAbbrev(*latestLedgerHeader));
             }
             else
             {
-                {
-                    LedgerTxn ltx(mApp.getLedgerTxnRoot());
-                    auto header = ltx.loadHeader();
-                    uint32_t ledgerVersion = header.current().ledgerVersion;
-                    if (mApp.getConfig().MODE_ENABLES_BUCKETLIST)
-                    {
-                        auto assumeStateWork =
-                            mApp.getWorkScheduler()
-                                .executeWork<AssumeStateWork>(has,
-                                                              ledgerVersion);
-                        if (assumeStateWork->getState() ==
-                            BasicWork::State::WORK_SUCCESS)
-                        {
-                            CLOG_INFO(Ledger,
-                                      "Assumed bucket-state for LCL: {}",
-                                      ledgerAbbrev(header.current()));
-                        }
-                        else
-                        {
-                            // Work should only fail during graceful shutdown
-                            releaseAssert(mApp.isStopping());
-                        }
-                    }
-                    advanceLedgerPointers(header.current());
-                }
-                handler();
+                // Work should only fail during graceful shutdown
+                releaseAssertOrThrow(mApp.isStopping());
             }
         }
-        else
-        {
-            LedgerTxn ltx(mApp.getLedgerTxnRoot());
-            advanceLedgerPointers(ltx.loadHeader().current());
-        }
+    }
+
+    // Step 3. Restore LedgerManager's internal state
+    advanceLedgerPointers(*latestLedgerHeader);
+
+    // In some modes, e.g. in-memory, core's state is rebuilt asynchronously via
+    // catchup. In this case, we're not able to load the network config at this
+    // time, and instead must let catchup do it when ready.
+    if (!isLedgerStateReady)
```

Comment:
I would rather change this to positive condition (i.e. `if (isLedgerStateReady) { loadConfig() } else { LOG(...) }` and get rid of the early return (it seems quite easy to forget about it and accidentally skip some logic independent of the config loading).

Reply:
ok, updated (fixed comments too)